src_c/line.c

```diff
@@ -490,6 +490,89 @@ pg_line_flip_ip(pgLineObject *self, PyObject *_null)
     Py_RETURN_NONE;
 }
 
+double 
+_lerp_helper(float start, float end, float amount) {
+    return (1 - amount) * start + amount * end;
+}
+
+static int
+_line_scale_helper(pgLineObject *line, double factor, double origin) {
+    if (factor == 1.0) {
+        return 1;
+    }
+    else if (factor == 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Cannot scale by 0");
+        return 0;
+    }
+    else if (factor < 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Cannot scale by a negative number");
+        return 0;
+    }
+
+    if (origin < 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Origin cannot be negative");
+        return 0;
+    }
+
+    double x1 = line->line.x1;
+    double y1 = line->line.y1;
+    double x2 = line->line.x2;
+    double y2 = line->line.y2;
+
+    double x1_factor = x1 * factor;
+    double y1_factor = y1 * factor;
+    double x2_factor = x2 * factor;
+    double y2_factor = y2 * factor;
+
+    double dx = _lerp_helper(x1_factor - x1, x2_factor - x2, origin);
+    double dy = _lerp_helper(y1_factor - y1, y2_factor - y2, origin);
```

Comment:
Could avoid a couple operations with this
```suggestion
    double fac_m_one = factor - 1;
    double dx = _lerp_helper(fac_m_one * x1, fac_m_one * x2, origin);
    double dy = _lerp_helper(fac_m_one * y1, fac_m_one * y2, origin);
```